File: BaseLibrary/command/equipment/GetMineGunsAnswerCommand.cs
using System.Collections.Generic;
using BaseLibrary.equip;

namespace BaseLibrary.command.equipment {
    public class GetMineGunsAnswerCommand : AEquipmentCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        /// <summary>
        /// Available mine guns to buy.
        /// </summary>
        /// <seealso cref="MineGun"/>
        public MineGun[] MINE_GUNS { get; private set; }

        public GetMineGunsAnswerCommand(MineGun[] mineGuns)
            : base() {
            MINE_GUNS = mineGuns;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/equipment/GetMineGunsAnswerCommand.java
package BaseLibrary.communication.command.equipment;

import BaseLibrary.communication.protocol.AProtocol;
import BaseLibrary.equipment.MineGun;

public class GetMineGunsAnswerCommand extends AEquipmentCommand {

	private static final String NAME = "MINE_GUNS_ANSWER";

	static {
		AProtocol.registerForDeserialize(NAME, GetMineGunsAnswerCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	private MineGun[] mineGuns;

	public GetMineGunsAnswerCommand(MineGun[] mineGuns) {
		setMineGuns(mineGuns);
	}

	/**
	 * Available mine guns to buy.
	 * 
	 * @see MineGun
	 */
	public MineGun[] getMineGuns() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return mineGuns;
	}

	private void setMineGuns(MineGun[] mineGuns) {
		pending = false;
		this.mineGuns = mineGuns;
	}
}
